media/videostitcher/stitcher_test.go

```diff
@@ -0,0 +1,146 @@
+		t.Fatalf("cloudresourcemanager.NewService: %v", err)
+	}
+	project, err := cloudresourcemanagerClient.Projects.Get(tc.ProjectID).Do()
+	if err != nil {
+		t.Fatalf("cloudresourcemanagerClient.Projects.Get.Do: %v", err)
+	}
+	projectNumber := strconv.FormatInt(project.ProjectNumber, 10)
+
+	testSlates(t, projectNumber, slateURI, updatedSlateURI)
+	t.Logf("\ntestSlates() completed\n")
+}
+
+// testSlates tests major operations on slates. Create, list, update,
+// and get operations check if the slate resource name is returned. The
+// delete operation checks for a hard-coded string response.
+func testSlates(t *testing.T, projectNumber, slateURI, updatedSlateURI string) {
+	tc := testutil.SystemTest(t)
+	buf := &bytes.Buffer{}
+
+	// Test setup
+
+	// Delete the default slate if it exists
+	if err := getSlate(buf, tc.ProjectID, location, slateID); err == nil {
+		testutil.Retry(t, 3, 2*time.Second, func(r *testutil.R) {
+			if err := deleteSlate(buf, tc.ProjectID, location, slateID); err != nil {
+				r.Errorf("deleteSlate got err: %v", err)
+			}
+		})
+	}
+
+	// Tests
+
+	// Create a new slate.
+	testutil.Retry(t, 3, 2*time.Second, func(r *testutil.R) {
+		slateName := fmt.Sprintf("projects/%s/locations/%s/slates/%s", projectNumber, location, slateID)
+		if err := createSlate(buf, tc.ProjectID, location, slateID, slateURI); err != nil {
+			r.Errorf("createSlate got err: %v", err)
+		}
+		if got := buf.String(); !strings.Contains(got, slateName) {
+			r.Errorf("createSlate got\n----\n%v\n----\nWant to contain:\n----\n%v\n----\n", got, slateName)
+		}
+	})
+	buf.Reset()
+
+	// List the slates for a given location.
+	testutil.Retry(t, 3, 2*time.Second, func(r *testutil.R) {
+		slateName := fmt.Sprintf("projects/%s/locations/%s/slates/%s", tc.ProjectID, location, slateID)
+		if err := listSlates(buf, tc.ProjectID, location); err != nil {
+			r.Errorf("listSlates got err: %v", err)
+		}
+		if got := buf.String(); !strings.Contains(got, slateName) {
+			r.Errorf("listSlates got\n----\n%v\n----\nWant to contain:\n----\n%v\n----\n", got, slateName)
+		}
+	})
+	buf.Reset()
+
+	// Update an existing slate.
+	testutil.Retry(t, 3, 2*time.Second, func(r *testutil.R) {
+		slateName := fmt.Sprintf("projects/%s/locations/%s/slates/%s", tc.ProjectID, location, slateID)
+		if err := updateSlate(buf, tc.ProjectID, location, slateID, updatedSlateURI); err != nil {
+			r.Errorf("updateSlate got err: %v", err)
+		}
+		if got := buf.String(); !strings.Contains(got, slateName) {
+			r.Errorf("updateSlate got\n----\n%v\n----\nWant to contain:\n----\n%v\n----\n", got, slateName)
+		}
+	})
+	buf.Reset()
+
+	// Get the updated slate.
+	testutil.Retry(t, 3, 2*time.Second, func(r *testutil.R) {
+		slateName := fmt.Sprintf("projects/%s/locations/%s/slates/%s", tc.ProjectID, location, slateID)
+		if err := getSlate(buf, tc.ProjectID, location, slateID); err != nil {
+			r.Errorf("getSlate got err: %v", err)
+		}
+		if got := buf.String(); !strings.Contains(got, slateName) {
```

Comment:
This should check that the update actually works.

Reply:
Added a test for this